pandas/tests/io/excel/test_odf.py

```diff
@@ -0,0 +1,76 @@
+import functools
+
+import numpy as np
+import pytest
+
+import pandas as pd
+import pandas.util.testing as tm
+
+pytest.importorskip("odf")
+
+
+@pytest.fixture(autouse=True)
+def cd_and_set_engine(monkeypatch, datapath):
+    func = functools.partial(pd.read_excel, engine="odf")
+    monkeypatch.setattr(pd, 'read_excel', func)
+    monkeypatch.chdir(datapath("io", "data"))
+
+
+def test_read_invalid_types_raises():
+    # the invalid_value_type.ods required manually editing
+    # of the included content.xml file
+    with pytest.raises(ValueError,
+                       match="Unrecognized type awesome_new_type"):
+        pd.read_excel("invalid_value_type.ods", header=None)
+
+
+def test_read_lower_diagonal():
+    # Make sure we can parse:
+    # 1
+    # 2 3
+    # 4 5 6
+    # 7 8 9 10
+
+    sheet = pd.read_excel("lowerdiagonal.ods", 'Sheet1',
+                          index_col=None, header=None)
+
+    assert sheet.shape == (4, 4)
```

Comment:
Might change this and subsequent tests below to use `tm.assert_frame_equal` in another iteration or PR